build.gradle

```diff
@@ -203,6 +205,59 @@ task wrapper(type: Wrapper) { gradleVersion = '1.12' }
 
 import javax.tools.ToolProvider
 
+task javadocTarball(type: Tar) {
+  baseName = "gobblin-javadoc-all"
+  destinationDir = new File(project.buildDir, baseName)
+  compression = Compression.GZIP
+  extension = 'tgz'
+  description = "Generates a tar-ball with all javadocs to ${destinationDir}/${archiveName}"
+}
+
+javadocTarball << {
+  def indexFile = new File(destinationDir, "index.md")
+  def version = rootProject.ext.javadocVersion
+  indexFile << """----
+layout: page
+title: Gobblin Javadoc packages ${version}
+permalink: /javadoc/${version}/
+----
+
+"""
+  rootProject.ext.javadocPackages.each {
+    indexFile << "* [${it}](${it})\n"
+  }
+}
+
+//Javadoc initialization for subprojects
+ext.javadocVersion = null != project.version ? project.version : "latest"
+if (ext.javadocVersion.indexOf('-') > 0) {
+  //remove any "-" addons from the version
+  ext.javadocVersion = javadocVersion.substring(0, javadocVersion.indexOf('-'))
+}
+
+ext.javadocPackages = new HashSet<String>()
+subprojects.each{Project pr -> 
+  if (file(pr.projectDir.absolutePath + "/src/main/java").exists()) {
+    rootProject.ext.javadocPackages += pr.name
+  }
+}
+
+subprojects {
+  plugins.withType(JavaPlugin) {
+ 
+    //Sometimes generating javadocs can lead to OOM. This may needs to be increased.
+    javadoc {
+      options.JFlags = ["-Xmx256m"]
+    }
+    
+    rootProject.tasks.javadocTarball.dependsOn project.tasks.javadoc
+    if ( rootProject.ext.javadocPackages.contains(project.name)) {
+      def cs = rootProject.tasks.javadocTarball.into(project.name){from(fileTree(dir: "${project.buildDir}/docs/javadoc/"))}
```

Comment:
The cs variable not, but the rest of the line adds the javadoc files to the common tarball.